op-node/rollup/derive/engine_update.go

```diff
@@ -116,6 +124,65 @@ func startPayload(ctx context.Context, eng ExecEngine, fc eth.ForkchoiceState, a
 	}
 }
 
+// makes parallel request to builder and engine to get the payload
+func getPayloadWithBuilderPayload(ctx context.Context, log log.Logger, eng ExecEngine, payloadInfo eth.PayloadInfo, l2head eth.L2BlockRef, builder builder.PayloadBuilder, metrics Metrics) (
+	*eth.ExecutionPayloadEnvelope, *eth.ExecutionPayloadEnvelope, error) {
+	// if builder is not enabled, return early with default path.
+	if !builder.Enabled() {
+		payload, err := eng.GetPayload(ctx, payloadInfo)
+		return payload, nil, err
+	}
+
+	log.Debug("requesting payload from builder", l2head.String(), "payloadInfo", payloadInfo)
+	ctxTimeout, cancel := context.WithTimeout(ctx, builder.Timeout())
+	defer cancel()
+	type result struct {
+		envelope *eth.ExecutionPayloadEnvelope
+		profit   *big.Int
+		error    error
+	}
+
+	ch := make(chan *result, 1)
+
+	// start the payload request to builder api
+	go func() {
+		start := time.Now()
+		payload, profit, err := builder.GetPayload(ctxTimeout, l2head, log)
+		metrics.RecordBuilderRequestTime(time.Since(start))
+		if err != nil {
+			ch <- &result{error: err}
+			return
+		}
+		ch <- &result{envelope: payload, profit: profit}
+	}()
+
+	envelope, err := eng.GetPayload(ctx, payloadInfo)
+	if err != nil {
+		log.Error("failed to get payload from engine", "error", err.Error())
+		return envelope, nil, err
+	}
+
+	// select the payload from builder if possible
+	select {
+	case <-ctxTimeout.Done():
+		metrics.RecordBuilderRequestTimeout()
+		metrics.RecordSequencerProfit(0, PayloadSourceEngine)
```

Comment:
this doesn't really make sense to me. Sequencer profit from source `PayloadSourceEngine` is always 0, is there need to record any profit from engine payloads at all? If trying to measure payload count it should already be covered by `RecordSequencerPayloadInserted`

Reply:
a more correct implementation would be the ability to get the block value or fees generated from the engine payload and record the profit here

Reply:
Yeah I agree. I thought profit meant the additional margin on top of base block reward. However I bet they would still be interested in measuring the additional profit from the builder so I'll just leave it as is and revisit later